Factory/OidcTokenResponseFactory.php

```diff
@@ -85,14 +95,15 @@ public function setMandatoryClaims(array $mandatoryClaims): self {
   }
 
   /**
-   * @param string $jwtTokenString
+   * @param string             $jwtTokenString
+   * @param OidcGrantInterface $grant
    *
    * @return OidcAccessTokenInterface
-   * @throws InvalidTokenException
+   * @throws InvalidAccessTokenException
    */
-  private function createAccessToken(string $jwtTokenString): OidcAccessTokenInterface {
+  private function createAccessToken(string $jwtTokenString, OidcGrantInterface $grant): OidcAccessTokenInterface {
     try {
-      $jwt = $this->createJWT($jwtTokenString, self::VALID_ACCESS_TOKEN_TYPES);
+      $jwt = $this->createJWT($jwtTokenString, $grant instanceof ClientCredentialsGrant ? OidcTokenInterface::TYPE_ACCESS_CLIENT_CREDENTIALS : OidcTokenInterface::TYPE_ACCESS);
```

Comment:
This works for now, but if we're getting more types, an array lookup or something similar might be better here :+1: